qiskit/mitigation/local_readout_mitigator.py

```diff
@@ -0,0 +1,269 @@
+        for i, ainv in enumerate(reversed(ainvs)):
+            einsum_args += [ainv.T, [self._num_qubits + i, i]]
+        einsum_args += [list(range(self._num_qubits, 2 * self._num_qubits))]
+        coeffs = np.einsum(*einsum_args).ravel()
+
+        expval = coeffs.dot(probs_vec)
+        stddev_upper_bound = self.stddev_upper_bound(shots, qubits)
+
+        return (expval, stddev_upper_bound)
+
+    def quasi_probabilities(
+        self,
+        data: Counts,
+        qubits: Optional[List[int]] = None,
+        clbits: Optional[List[int]] = None,
+        shots: Optional[bool] = False,
+    ) -> QuasiDistribution:
+        """Compute mitigated quasi probabilities value.
+
+        Args:
+            data: counts object
+            qubits: qubits the count bitstrings correspond to.
+            clbits: Optional, marginalize counts to just these bits.
+            shots: the number of shots.
+
+        Returns:
+            QuasiDistibution: A dictionary containing pairs of [output, mean] where "output"
+                is the key in the dictionaries,
+                which is the length-N bitstring of a measured standard basis state,
+                and "mean" is the mean of non-zero quasi-probability estimates.
+
+        Raises:
+            QiskitError: if qubit and clbit kwargs are not valid.
+        """
+        if qubits is None:
+            qubits = self._qubits
+
+        num_qubits = len(qubits)
+
+        probs_vec, shots = counts_probability_vector(data, clbits=clbits, qubits=qubits)
+
+        # Get qubit mitigation matrix and mitigate probs
+        ainvs = self._mitigation_mats[list(qubits)]
+
+        # Apply transpose of mitigation matrix
+        prob_tens = np.reshape(probs_vec, num_qubits * [2])
+        einsum_args = [prob_tens, list(range(num_qubits))]
+        for i, ainv in enumerate(reversed(ainvs)):
+            einsum_args += [ainv, [num_qubits + i, i]]
+        einsum_args += [list(range(num_qubits, 2 * num_qubits))]
+        probs_vec = np.einsum(*einsum_args).ravel()
+
+        probs_dict = {}
+        for index, _ in enumerate(probs_vec):
+            probs_dict[index] = probs_vec[index]
+
+        quasi_dist = QuasiDistribution(probs_dict)
+        quasi_dist._stddev_upper_bound = self.stddev_upper_bound(shots, qubits)
+
+        return quasi_dist
+
+    def mitigation_matrix(self, qubits: List[int] = None) -> np.ndarray:
+        r"""Return the measurement mitigation matrix for the specified qubits.
+        The mitigation matrix :math:`A^{-1}` is defined as the inverse of the
+        :meth:`assignment_matrix` :math:`A`.
+        Args:
+            qubits: Optional, qubits being measured for operator expval.
+        Returns:
+            np.ndarray: the measurement error mitigation matrix :math:`A^{-1}`.
+        """
+        if qubits is None:
+            qubits = self._qubits
+        if isinstance(qubits, int):
+            qubits = [qubits]
+        mat = self._mitigation_mats[qubits[0]]
```

Comment:
I think this needs to be position of the qubit in `self._qubits`

Reply:
Fixed